### PR TITLE
RDoc-1794-tryconvert-index

### PR DESCRIPTION
**Pull request by morhilai**
Documented `TryConvert()` and `tryConvertToNumber()` in new page

### CODE REVIEW

File: Documentation/5.0/Raven.Documentation.Pages/indexes/number-type-conversion.dotnet.markdown [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+{CODE-BLOCK:csharp}
+protected T? TryConvert<T>(object value)
+{CODE-BLOCK/}
+
+| Parameter | Type | Description |
+| - | - | - |
+| **T** | Generic type parameter | The numerical type to which you want to convert your value. Possible values:<br/>- `int`<br/>- `long`<br/>- `float`<br/>- `double` |
+| **value** | `object` | The value you want to convert, such as a document field. If you pass a `string` or `object`, the method will attempt to parse them for a numerical value. |
+
+In JavaScript syntax, use `tryConvertToNumber()`.
```
Comment (ppekrol) [inline Documentation/5.0/Raven.Documentation.Pages/indexes/number-type-conversion.dotnet.markdown:38]: The JS syntaxt is only available in 5.1. We need to apply that change there.

File: Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/Indexes/NumberTypeConversion.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    {
+        #region tryconvert_linq
+        public class TryConvertIndexLINQ : AbstractIndexCreationTask<Item>
+        {
+            public TryConvertIndexLINQ()
+            {
+                Map = items => from item in items
+                               select new
+                               {
+                                   DoubleValue = TryConvert<double>(item.DoubleValue) ?? -1,
```
Comment (ppekrol) [inline Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/Indexes/NumberTypeConversion.cs:20]: Index like that is good for writing a test, but it does let's have something more meaningful e.g.

Employees_ByPostalCode
```
from employee in employees
select
{
    PostalCode = TryConvert<long>(employee.Address.PostalCode) ?? -1
}
```

Then write in the text:
In order to find all the employees with an invalid postal code you can issue following query:
```
List<Employee> employeesWithoutPostalCode = session
    .Query<Employees_ByPostalCode.Result, Employees_ByPostalCode>()
    .Where(x => x.PostalCode == -1)
    .OfType<Employee>()
    .ToList();
```

File: Documentation/5.0/Raven.Documentation.Pages/indexes/number-type-conversion.dotnet.markdown (hunk 1 above)
Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/indexes/number-type-conversion.dotnet.markdown]: I added an info box saying "Available from version 5.1 and later" - is that what you meant?

If so, fixed.

File: Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/Indexes/NumberTypeConversion.cs (hunk 2 above)
Comment (morhilai) [reply Documentation/5.0/Samples/csharp/Raven.Documentation.Samples/Indexes/NumberTypeConversion.cs:21]: Done

File: Documentation/5.0/Raven.Documentation.Pages/indexes/number-type-conversion.dotnet.markdown [hunk 3]
```diff
@@ -33,17 +33,21 @@ protected T? TryConvert<T>(object value)
...
 | **T** | Generic type parameter | The numerical type to which you want to convert your value. Possible values:<br/>- `int`<br/>- `long`<br/>- `float`<br/>- `double` |
-| **value** | `object` | The value you want to convert, such as a document field. If you pass a `string` or `object`, the method will attempt to parse them for a numerical value. |
+| **value** | `object` | The value you want to convert, such as a document field. If you pass a `string` or `object`, the method will attempt to parse it for a numerical value. |
 
-In JavaScript syntax, use `tryConvertToNumber()`.
+{SAFE: }
+Available from version 5.1 and later  
+{SAFE/}
+
+In **JavaScript syntax**, use `tryConvertToNumber()`.
```
Comment (ppekrol) [inline Documentation/5.0/Raven.Documentation.Pages/indexes/number-type-conversion.dotnet.markdown:42]: You need to copy that article to 5.1 docs and document JS side there only.

Comment (morhilai) [reply Documentation/5.0/Raven.Documentation.Pages/indexes/number-type-conversion.dotnet.markdown]: Done

**Pull request merged**